system-test/datastore.ts

```diff
@@ -830,69 +824,51 @@ describe('Datastore', () => {
       });
     });
 
-    it('should commit all saves and deletes at the end', done => {
+    it('should commit all saves and deletes at the end', async () => {
       const deleteKey = datastore.key(['Company', 'Subway']);
       const key = datastore.key(['Company', 'Google']);
       const incompleteKey = datastore.key('Company');
 
-      datastore.save(
+      await datastore.save(
         {
           key: deleteKey,
           data: {},
-        },
-        err => {
-          assert.ifError(err);
-
-          const transaction = datastore.transaction();
+        }
+      );
+      const transaction = datastore.transaction();
 
-          transaction.run(err => {
-            assert.ifError(err);
+      await transaction.run();
+      transaction.delete(deleteKey);
 
-            transaction.delete(deleteKey);
+      transaction.save([
+        {
+          key,
+          data: {rating: 10},
+        },
+        {
+          key: incompleteKey,
+          data: {rating: 100},
+        },
+      ]);
 
-            transaction.save([
-              {
-                key,
-                data: {rating: 10},
-              },
-              {
-                key: incompleteKey,
-                data: {rating: 100},
-              },
-            ]);
+      await transaction.commit();
 
-            transaction.commit(err => {
-              assert.ifError(err);
+      // Incomplete key should have been given an ID.
+      assert.strictEqual(incompleteKey.path.length, 2);
 
-              // Incomplete key should have been given an ID.
-              assert.strictEqual(incompleteKey.path.length, 2);
-
-              async.parallel(
-                [
-                  // The key queued for deletion should have been deleted.
-                  callback => {
-                    datastore.get(deleteKey, (err, entity) => {
-                      assert.ifError(err);
-                      assert.strictEqual(typeof entity, 'undefined');
-                      callback();
-                    });
-                  },
+      await Promise.all([
+        // Deletes the key that is in the deletion queue.
+        datastore.get(deleteKey, (err, entity) => {
```

Comment:
Mixing callbacks and promises. Use a `.then` to do the assertion.

Reply:
Done! https://github.com/googleapis/nodejs-datastore/pull/272/commits/c120a83d74bdb5a4c96e361ab7cf40b4684839bc